src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.Ssl.cs

```diff
@@ -64,9 +80,113 @@ internal static string GetProtocolVersion(SafeSslHandle ssl)
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern SafeSharedX509StackHandle SslGetPeerCertChain(SafeSslHandle ssl);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int SslCtxUseCertificate(SafeSslContextHandle ctx, SafeX509Handle certPtr);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int SslCtxUsePrivateKey(SafeSslContextHandle ctx, SafeEvpPKeyHandle keyPtr);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int SslCtxCheckPrivateKey(SafeSslContextHandle ctx);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern int BioCtrlPending(SafeBioHandle bio);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SslCtxSetQuietShutdown(SafeSslContextHandle ctx);
+
+        [DllImport(Libraries.CryptoNative, EntryPoint = "SslGetClientCAList")]
+        private static extern SafeSharedX509NameStackHandle SslGetClientCAListPrivate(SafeSslHandle ssl);
+
+        internal static SafeSharedX509NameStackHandle SslGetClientCAList(SafeSslHandle ssl)
+        {
+            Interop.Crypto.CheckValidOpenSslHandle(ssl);
+
+            SafeSharedX509NameStackHandle handle = SslGetClientCAListPrivate(ssl);
+
+            if (!handle.IsInvalid)
+            {
+                handle.SetParent(ssl);
+            }
+
+            return handle;
+        }
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SslCtxSetVerify(SafeSslContextHandle ctx, [MarshalAs(UnmanagedType.FunctionPtr)] SslCtxSetVerifyCallback callback);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SetEncryptionPolicy(SafeSslContextHandle ctx, EncryptionPolicy policy);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void SslCtxSetClientCAList(SafeSslContextHandle ctx, SafeX509NameStackHandle x509NameStackPtr);
+
+        [DllImport(Libraries.CryptoNative)]
+        internal static extern void GetStreamSizes(out int header, out int trailer, out int maximumMessage);
+
+        internal static class SslMethods
+        {
+            internal static readonly IntPtr TLSv1_method = TlsV1Method();
+            internal static readonly IntPtr TLSv1_1_method = TlsV1_1Method();
+            internal static readonly IntPtr TLSv1_2_method = TlsV1_2Method();
+            internal static readonly IntPtr SSLv3_method = SslV3Method();
+            internal static readonly IntPtr SSLv23_method = SslV2_3Method();
+
+#if DEBUG
+            static SslMethods()
+            {
+                Debug.Assert(TLSv1_method != IntPtr.Zero, "TLSv1_method is null");
+                // TLSv1_1_method and TLSv1_2_method do not exist on earlier versions of OpenSSL
+                // Debug.Assert(TLSv1_1_method != IntPtr.Zero, "TLSv1_1_method is null");
+                // Debug.Assert(TLSv1_2_method != IntPtr.Zero, "TLSv1_2_method is null");
```

Comment:
Is there any validation we can do for these, such as checking the SSL version and asserting that we successfully got these if the version is new enough?